components/mailgun/sources/common-webhook.js

```diff
@@ -0,0 +1,97 @@
+const mailgun = require("../mailgun.app");
+
+module.exports = {
+  props: {
+    mailgun,
+    webhookSigningKey: {
+      type: "string",
+      secret: true,
+      label: "Mailgun webhook signing key",
+      description:
+        "Your Mailgun webhook signing key, found [in your Mailgun dashboard](https://app.mailgun.com/app/dashboard), located under Settings on the left-hand nav and then in API Keys look for webhook signing key. Required to compute the authentication signature of events."
+    },
+    http: "$.interface.http",
+    db: "$.service.db",
+  },
+  methods: {
+    verify(signingKey, timestamp, token, signature) {
+      const crypto = require("crypto");
+      const encodedToken = crypto
+        .createHmac("sha256", signingKey)
+        .update(timestamp.concat(token))
+        .digest("hex");
+      return encodedToken === signature;
+    },
+    emitEvent(eventPayload) {
+      const eventTypes = this.getEventType();
```

Comment:
Can you add a `getEventType` method stub here that throws a "not implemented" error? e.g. see https://github.com/PipedreamHQ/pipedream/blob/7f03b49859616f3576d0c3d24d24816560962770/components/snowflake/sources/common.js#L67

Reply:
Yes, it was added.